app/controllers/api/v1/questions_controller.rb

```diff
@@ -0,0 +1,136 @@
+
+  # Show method returns the question object with id - {:id}
+  # GET on /questions/:id
+  def show
+    begin
+      @question = Question.find(params[:id])
+      render json: @question, status: :ok
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found
+    end
+  end
+
+  # Create method returns the JSON object of created question
+  # POST on /questions
+  def create
+    begin
+      questionnaire_id = params[:questionnaire_id] unless params[:questionnaire_id].nil?
+      num_of_existed_questions = Questionnaire.find(questionnaire_id).questions.size
+      question = Question.create(
+        txt: params[:txt],
+        questionnaire_id: questionnaire_id,
+        seq: num_of_existed_questions + 1,
+        question_type: params[:question_type],
+        break_before: true)
+      case question.question_type
+        when 'Scale'
+          question.weight = params[:weight]
+          question.max_label = 'Strongly agree'
+          question.min_label = 'Strongly disagree'
+        when 'Cake', 'Criterion'
+          question.weight = params[:weight]
+          question.max_label = 'Strongly agree'
+          question.min_label = 'Strongly disagree'
+          question.size = '50, 3'
+        when 'Dropdown'
+          question.alternatives = '0|1|2|3|4|5'
+          question.size = nil
+        when 'TextArea'
+          question.size = '60, 5'
+        when 'TextField'
+          question.size = '30'
+      end
+    
+      question.save!
+      render json: question, status: :created
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return  
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Destroy method deletes the question with id - {:id}
+  # DELETE on /questions/:id
+  def destroy
+    begin
+      @question = Question.find(params[:id])
+      @question.destroy
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found and return
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # show_all method returns all questions associated to a questionnaire with id - {:id}
+  # GET on /questions/show_all/questionnaire/:id
+  def show_all
+    begin
+      @questionnaire = Questionnaire.find(params[:id])
+      @questions = @questionnaire.questions
+      render json: @questions, status: :ok
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found
+    rescue ActiveRecord::RecordInvalid
+      render json: $ERROR_INFO.to_s, status: :unprocessable_entity
+    end
+  end
+
+  # Delete_all method deletes all the questions and returns the status code
+  # DELETE on /questions/delete_all/questionnaire/<questionnaire_id>
+  # Endpoint to delete all questions associated to a particular questionnaire.
+  def delete_all
```

Comment:
```ruby
def delete_all
  begin
    questionnaire = Questionnaire.find(params[:id])
    if questionnaire.questions.empty?
      render json: "No questions associated with questionnaire ID #{params[:id]}.", status: :unprocessable_entity and return
    end

    questionnaire.questions.destroy_all
    render json: "All questions for questionnaire ID #{params[:id]} have been deleted.", status: :ok
  rescue ActiveRecord::RecordNotFound
    render json: "Questionnaire ID #{params[:id]} not found.", status: :not_found and return
  end
end
```

- removed the unnecessary instance variable @questionnaire, since we are only using it once.
- removed the msg variable and inline its value into the render method call.
- used empty? instead of size > 0 to check if the questions array is empty.
- reorderd the render calls to always be at the end, which is a common convention in Rails.
- simplified the error messages to make them more concise and readable.

Reply:
Commit e86e810 addresses this review.